crm_lead_code/__init__.py

```diff
@@ -0,0 +1,25 @@
+##############################################################################
+# For copyright and license notices, see __manifest__.py file in root directory
+##############################################################################
+
+from . import models
+from odoo import api, SUPERUSER_ID
+
+
+def create_code_equal_to_id(cr):
+    cr.execute('ALTER TABLE crm_lead '
+               'ADD COLUMN code character varying;')
+    cr.execute('UPDATE crm_lead '
+               'SET code = id;')
+
+
+def assign_old_sequences(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, dict())
+    lead_obj = env['crm.lead']
+    sequence_obj = env['ir.sequence']
+    leads = lead_obj.search([], order="id")
+    for lead_id in leads.ids:
```

Comment:
Why using sql query Here?

```python
    for lead in leads:
         lead.code = sequence_obj.next_by_code('crm.lead')
```

Reply:
Since I only migrated the module and didn't rewrite it, I tried to change as little as possible to make future bugfixes etc easier in all versions.  In my opinion I would only change things if they bring a clear advantage or are necessary due to Odoo version changes.  Where do you see the advantage of not using SQL directly but using the ORM?  In this case performance doesn't matter as it is executed only once, but this way you make sure that there are no unwanted side effects through the ORM. Like setting the write_date and write_uid.

Reply:
@sbejaoui this was done by SQL on purpose, as we want to avoid triggering any ORM code assigning this, as it's an initialization. This is also done for performance reasons.

Reply:
> Where do you see the advantage of not using SQL directly but using the ORM?

IMO, any use of sql queries should be justified otherwise the ORM is here to abstract the sql level and should be used in all cases by default.

I think that the sql query here can be justified by the need of avoiding `write_uid` update and there in not any computed field depends on this one.